### DISCUSSION TITLE
Make session expiration message localizable

### DISCUSSION THREAD

**Discussion by dakira** [livewire/livewire#1338] [3. Feature ideas]
The code that handles expired sessions shows a javasscript confirm dialog with the message *""This page has expired due to inactivity. Would you like to refresh the page?""*

This English message poses a problem to users who don't understand English. ¯\_(ツ)_/¯

This is hardcoded in http.js and thus in the minified livewire.js. So it's rather hard to localize I guess. If instead the html modal with the response was shown, it would be easier to show a localized message with the correct response to the users (I'm pretty sure it was like this in earlier versions of Livewire).

I'm not sure which way is the best to go. Load the text from a localizable html file or maybe revert the file back to the state before this commit: https://github.com/livewire/livewire/commit/88fc8abf24bcdb590e484dd84783ec09d1f60e1c.

**Comment by ChristofK** [comment]
I would also love this feature.
I solved this problem for now by hardcoding the translation to the js file. Since my project is only in two languages this works for me.

**Comment by phanan** [comment]
Somehow this doesn't seem to receive much love from the maintainers. I believe it's a legit request though, knowing that more and more international users are adopting Livewire. Not to mention, it's generally not the best idea to hardcode these strings in a library.

I'd be more than happy to take a look at this. However, since I'm not familiar with Livewire's internals yet, I'd need some initial cue. 

/pinging @calebporzio.

**Comment by enigmatic-user** [comment]
I just stumbled upon this, too, so: +1.

**Comment by RobertCordes** [comment]
Yeah, one of my clients just called and told me about this. They were worried they did something wrong, because they didn't understand the message…

**Comment by dakira** [comment]
Hey @calebporzio, if you have some ideas on how to approach this, I'd be happy to work on a PR (at least I would try). ;-)
